File: Data Structures & Algorithms/add-two-numbers/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        carry = 0
        res = ListNode(0)
        cur = res

        while l1 is not None or l2 is not None or carry:
            digit1 = l1.val if l1 else 0
            digit2 = l2.val if l2 else 0
            total = digit1 + digit2 + carry

            carry = total // 10

            cur.next = ListNode(total % 10)
            cur = cur.next

            if l1 :
                l1 = l1.next
            if l2 :
                l2 = l2.next

        return res.next

        

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] res = new int[nums.length];

        for (int i = 0; i < nums.length; i++){
            int num = 1;
            for (int j = 0; j < nums.length; j++){
                if (j != i) {
                    num *= nums[j];
                }
            }
            res[i] = num;
        }

        return res;
    }
}  
